test/security/hex.js

```diff
@@ -1,148 +1,78 @@
-  it("should differentiate between a list of 0 items and a list of one 0", function(done) {
+  it("should differentiate between a list of 0 items and a list of one 0", function() {
     assert.notStrictEqual(to.rpcDataHexString(Buffer.from("", "hex")), to.rpcDataHexString(Buffer.from("00", "hex")));
-    done();
   });
 });
 
-function noLeadingZeros(method, result, path) {
-  if (!path) {
-    path = "result";
-  }
-
-  if (typeof result === "string") {
-    if (/^0x/.test(result)) {
-      const asHex = to.rpcQuantityHexString(result);
-      assert.strictEqual(result, asHex, `Field ${path} in ${method} response has leading zeroes.`);
-    }
-  } else if (typeof result === "object") {
-    for (var key in result) {
-      if (result.hasOwnProperty(key)) {
-        if (Array.isArray(result)) {
-          path += [key];
-        } else {
-          path += "." + key;
-        }
-        noLeadingZeros(method, result[key], path + (path ? "." : "") + key);
-      }
-    }
-  }
-}
-
-describe("JSON-RPC Response", function() {
-  var web3 = new Web3();
-  var provider = Ganache.provider();
-  web3.setProvider(provider);
-
-  var accounts;
-  before(function(done) {
-    web3.eth.getAccounts(function(err, accs) {
-      if (err) {
-        return done(err);
-      }
-      accounts = accs;
-      done();
-    });
+describe("to.hex", function() {
+  it("should print '0x' for input '' (blank)", function() {
+    assert.strictEqual(to.hex(Buffer.from("")), "0x");
   });
+});
 
+describe.skip("JSON-RPC Response", function() {
   // skipping this test for now as they aren't verifying the right thing that
   // is, leading zeroes are fine in some response fields. we need a better model
   // of expected response formatting/padding.
-  it.skip("should not have leading zeros in rpc quantity hex strings", function(done) {
-    var request = {
-      jsonrpc: "2.0",
-      method: "eth_getTransactionCount",
-      params: [accounts[0], "pending"],
-      id: 1
-    };
+  it("should not have leading zeros in rpc quantity hex strings", async function() {
+    const { accounts, web3 } = await initializeTestProvider();
 
-    provider.sendAsync(request, function(err, result) {
-      if (err) {
-        return done(err);
-      }
-
-      noLeadingZeros("eth_getTransactionCount", result);
-
-      request = {
-        jsonrpc: "2.0",
-        method: "eth_sendTransaction",
-        params: [
-          {
-            from: accounts[0],
-            to: accounts[1],
-            value: "0x100000000"
-          }
-        ],
-        id: 2
-      };
+    let method = "eth_getTransactionCount";
+    let params = [accounts[0], "pending"];
 
-      provider.sendAsync(request, function() {
-        // Ignore eth_sendTransaction result, it returns the transaction hash.
-        // A transaction hash is a 'DATA' type, which can have leading zeroes
-        // to pad it to an even string length (4 bit per char, so whole bytes).
+    let result = send(method, params, web3);
```

Comment:
`send` needs to be `await`ed, right?

Reply:
@ccowell 	☝️

Reply:
Good catch but we skip this test. Should we keep it?  @davidmurdoch

Reply:
Yes, we should keep it.